src/components/AudioResultsList.vue

```diff
@@ -56,19 +57,24 @@ export default {
     },
   },
   async fetch() {
-    if (!this.$store.state.audios.length) {
-      await this.$store.dispatch(FETCH_MEDIA, {
+    if (!this.audios.length) {
+      await this.fetchMedia({
         ...this.$store.state.query,
         mediaType: AUDIO,
       })
     }
   },
   computed: {
-    audios() {
-      return this.$store.state.audios
-    },
+    ...mapState({
+      audios: (state) => state.audios,
```

Comment:
The strings do not work for namespaced stores, so for consistency I used the same syntax everywhere.
I thought that only Webstorm for some reason doesn't like dots in mapState (and warns about them), but apparently it's a Vuex limitation, see https://github.com/vuejs/vuex/issues/459 and https://github.com/vuejs/vuex/issues/1592

Reply:
For namespaced stores, the syntax is to use `/`, not `.` as the separator.

Reply:
I tried out different versions of state mapping in both WebStorm and Visual Studio Code, here are some results I get:

- ``...mapState(REPORT_CONTENT, { isReportFormVisible: 'isReportFormVisible' })``
(+) Clear, easy to set for namespaced modules.
(-) Error-prone: No autocomplete in either WebStorm or Visual Studio Code, the property is not a constant
(-) Webstorm shows a warning about 'isReportFormVisible': "Cannot resolve Vuex state isReportFormVisible"
(-) Not clear if it works for nested objects

- `...mapState(REPORT_CONTENT, {  
  isReportFormVisible: (state) => state.isReportFormVisible,  
}),`
(-) Verbose
(+) Easier to avoid errors:  both WebStorm and Visual Studio Code autocomplete the property after the dot (`isReportFormVisible`)
(+) No warnings.

- ``...mapState(REPORT_CONTENT, ['isReportFormVisible'])``
(+) Short
(-) Error prone: No autocomplete in either WebStorm or Visual Studio Code, the property is not a constant
(-) WebStorm warns that it "Cannot resolve Vuex state isReportFormVisible".
(-) Does not work well for nested objects (such as the `query` object in the search store)